### PR TITLE
50% Functions Decompiled

### PR DESCRIPTION
**Pull request by ethteck**
Assorted decomp, and now 50% of the functions in the game have been decompiled!

### CODE REVIEW

File: include/common_structs.h [hunk 1]
```diff
@@ -315,23 +315,27 @@ typedef struct ScriptInstance {
 
 typedef ScriptInstance* ScriptList[MAX_SCRIPTS];
 
+struct Entity;
```
Comment (nanaian) [inline include/common_structs.h:318]: 🤔

File: include/common_structs.h [hunk 2]
```diff
@@ -1125,15 +1129,18 @@ typedef struct PartnerAnimations {
...
 typedef struct Shadow {
     /* 0x00 */ s32 flags;
-    /* 0x04 */ char unk_04[2];
+    /* 0x04 */ u8 listIndex;
+    /* 0x05 */ u8 unk_05;
     /* 0x06 */ u8 unk_06;
     /* 0x07 */ char unk_07;
-    /* 0x08 */ s16 unk_08; // entity model index?
-    /* 0x0A */ char unk_0A[6];
+    /* 0x08 */ s16 entityModelID; // entity model index?
```
Comment (nanaian) [inline include/common_structs.h:1136]: Remove this comment now pls

File: src/102610_len_2330.c [hunk 3]
```diff
@@ -61,20 +60,16 @@ void func_802E1204(Entity* entity) {
...
-    s32* temp1 = &D_802EB390;
-    s32* temp2 = &D_802EB394;
-    *temp1 = 0;
-    *temp2 = 0;
-    D_802EB39C = load_string(0x1D0004, temp2);
-    D_802EB398 = load_string(0x1E000A, temp1);
+    D_802EB390 = 0;
+    D_802EB394 = 0;
+    D_802EB39C = load_string(0x1D0004, &D_802EB394);
+    D_802EB398 = load_string(0x1E000A, &D_802EB390);
```
Comment (nanaian) [inline src/102610_len_2330.c:72]: We should parameterise `0x1D0000`, `0x1D0004`, and `0x1E000A` via `message_ids.h` (see also its configuration, `tools/splat_ext/msg.yaml`). I don't mind if you don't do that in this PR, but it'd be preferable 😛

Here are the messages (which also give nice names for the unnamed `D_` variables and the functions themselves):
```
#message:1D:000 {
	[Style inspect]
	Hit this block to save your[BR]
	game progress.[BR]
	[Wait][Next]
	When the game is over, your[BR]
	status will return to what it[BR]
	was the last time you saved,[BR]
	and you can continue to play.[BR]
	[Wait][Next]
	When you find this block along [BR]
	the way, remember to hit it[BR]
	to save![Wait][End]
}

#message:1D:004 {
	[Style inspect]
	Do you want to save?[BR]
	[Yield][End]
}

#message:1E:00A {
	[Style choice pos=110,90 size=115,46]
	[DelayOff]
	[Cursor 0][Option 0]Save[BR]
	[Cursor 1][Option 1]Don't save[BR]
	[Option 255][DelayOn]
	[SetCancel 1][EndChoice 2][End]
}
```

File: src/102610_len_2330.c [hunk 4]
```diff
@@ -83,16 +78,13 @@ void func_802E12F8(void) {
...
-    if (*temp != 0) {
+    if (D_802EB394 != 0) {
         exec_entity_updatecmd(entity);
     }
 }
 
 void func_802E1350(Entity* entity) {
-    s32* temp = &D_802EB390;
-    if (*temp != 0) {
+    if (D_802EB390 != 0) {
```
Comment (nanaian) [inline src/102610_len_2330.c:87]: This and `D_802EB394` are bools, so we can drop the `!= 0`.

File: src/a5dd0_len_114e0.c [hunk 5]
```diff
@@ -18,32 +18,467 @@ typedef struct RenderTaskEntry {
...
+
+void update_entities(void) {
+    s32 i;
 
-INCLUDE_ASM(s32, "a5dd0_len_114e0", update_entities);
+    D_801512BC = 0;
+    D_801512C0 = 0;
+    D_80151330++;
+
+    for (i = 0; i < 30; i++) {
```
Comment (nanaian) [inline src/a5dd0_len_114e0.c:329]: `30` can be `MAX_ENTITIES`. Awesome match btw!

File: src/a5dd0_len_114e0.c [hunk 6]
```diff
@@ -18,32 +18,467 @@ typedef struct RenderTaskEntry {
...
+    update_shadows();
+    D_8015A578 = 0;
+}
+
+void update_shadows(void) {
+    s32 i;
+
+    D_80151324 = 0;
+
+    for (i = 0; i < 60; i++) {
```
Comment (nanaian) [inline src/a5dd0_len_114e0.c:448]: `MAX_SHADOWS`

File: src/a5dd0_len_114e0.c [hunk 7]
```diff
@@ -64,15 +499,134 @@ void entity_model_set_shadow_color(s32 alpha) {
     gDPSetPrimColor(gMasterGfxPos++, 0, 0, 0x00, 0x00, 0x00, alpha);
 }
 
-INCLUDE_ASM(s32, "a5dd0_len_114e0", render_entities);
+void render_entities(void) {
+    s32 phi_v0;
+    s32 i;
+
+    for (i = 0; i < 30; i++) {
```
Comment (nanaian) [inline src/a5dd0_len_114e0.c:506]: `MAX_ENTITIES`

File: src/a5dd0_len_114e0.c [hunk 8]
```diff
@@ -64,15 +499,134 @@ void entity_model_set_shadow_color(s32 alpha) {
...
+    }
+
+    render_shadows();
+}
+
+void render_shadows(void) {
+    s32 i;
 
-INCLUDE_ASM(s32, "a5dd0_len_114e0", render_shadows);
+    for (i = 0; i < 60; i++) {
```
Comment (nanaian) [inline src/a5dd0_len_114e0.c:582]: `MAX_SHADOWS`

File: src/evt/map_api.c [hunk 9]
```diff
@@ -351,9 +392,47 @@ ApiStatus UpdateColliderTransform(ScriptInstance* script, s32 isInitialCall) {
...
+    if (unkStruct->id2 >= 0) {
+        set_zone_enabled(unkStruct->id2, enabled);
+    }
 
-INCLUDE_ASM(ApiStatus, "evt/map_api", SetZoneEnabled, ScriptInstance* script, s32 isInitialCall);
+    switch (enabled) {
+        case 0:
+            unkStruct->flags |= 0x10000;
+            break;
+        case 1:
```
Comment (nanaian) [inline src/evt/map_api.c:429]: Maybe use `TRUE` and `FALSE` for these cases. Interesting they used a switch, lol

**Review by nanaian — CHANGES_REQUESTED** [review]
GG. Few small things

File: include/common_structs.h (hunk 1 above)
Comment (nanaian) [reply include/common_structs.h:318]: Why is this needed? Doesn't `struct Entity` suffice or does cc hate us

File: src/102610_len_2330.c (hunk 4 above)
Comment (nanaian) [reply src/102610_len_2330.c:87]: Also elsewhere -- e.g. `D_802EB390 = 0` up above -- can be changed to using bool defs too

File: include/common_structs.h (hunk 1 above)
Comment (ethteck) [reply include/common_structs.h:318]: It prevents a really nasty warning that comes up all over the place, saying that the `struct Entity` in the function typedef is limited to only that scope or something

**Pull request merged**